python/tests/test_DFtoVW.py

```diff
@@ -31,23 +32,50 @@ def test_feature_column_renaming_and_tag():
     assert first_line == "1 id_1| col_x:2"
 
 
-def test_feature_value_with_empty_name():
-    df = pd.DataFrame({"idx": ["id_1"], "y": [1], "x": [2]})
-    conv = DFtoVW(
-        label=SimpleLabel("y"),
-        tag="idx",
-        features=Feature(value="x", rename_feature=""),
-        df=df,
-    )
-    first_line = conv.convert_df()[0]
-    assert first_line == "1 id_1| :2"
-
-
 def test_multiple_lines():
     df = pd.DataFrame({"y": [1, -1], "x": [1, 2]})
-    conv = DFtoVW(label=SimpleLabel("y"), features=Feature(value="x"), df=df,)
+    conv = DFtoVW(label=SimpleLabel("y"), features=Feature(value="x"), df=df)
     lines_list = conv.convert_df()
-    assert lines_list == ["1 | 1", "-1 | 2"]
+    assert lines_list == ["1 | x:1", "-1 | x:2"]
+
+
+def test_empty_rename_feature():
+    df = pd.DataFrame({"a":[1]})
+    first_line = DFtoVW(df=df, features=Feature("a", rename_feature="")).convert_df()[0]
+    assert  first_line == "| :1"
+
+
+def test_mixed_type_features():
+    df = pd.DataFrame({"y": [1], "x1": ["a"], "x2": [2]})
+    conv = DFtoVW(label=SimpleLabel("y"),
+                  features=[Feature(value=colname) for colname in ["x1", "x2"]],
+                  df=df)
+    first_line = conv.convert_df()[0]
+    assert first_line == "1 | x1=a x2:2"
+
+
+def test_as_type_in_features():
```

Comment:
Can we still determine a different column to hold the feature values?
i.e. what if we want `"a"` to be categorical but also have different values other than 1, `a=2:4` for example

Reply:
Unfortunately it was disabled in #2601. The title of the PR is "Remove the possibility in DFtoVW to provide constant values" but actually it's more "Allow just one type of value (const or _Col) in DFtoVW classes". What was possible before is to give in the `name` arg the column name and set arg `name_from_df=True` to use the values of the column instead of constant string name.

Could you give an example of why you would do this (setting specific feature values to a categorical column), the use case is not really clear for me ? Thanks

Reply:
I synced with @lalo off line, it seems that the real world use case for categorical features having values other than `1` is probably not worth supporting and if someone really wants to do that explicitly then they would probably specify `a=2` as the column name